scripts/update_files.py

```diff
@@ -0,0 +1,46 @@
+from pathlib import Path
+import re
+import os
+import sys
+
+def check_tex_file_exists(repo_name: str, base_path: Path):
+    """Check if a .tex file with the same name as the repository exists."""
+    tex_file_path = base_path / f"{repo_name}.tex"
+    if not tex_file_path.exists():
+        msg = f"No .tex file found with the name {tex_file_path.name}. File name must match repo name."
+        raise FileNotFoundError(msg)
+
+def update_readme_content(content: str, repo_name: str, github_repository: str) -> str:
+    """Updates the README.md content."""
+    # Update the first line with the repository name
+    content = re.sub(r'^.*', f'# {repo_name}', content, count=1)
```

Comment:
Can we make this more robust and not depend on the header being line 1. Also try to avoid the comments in line 15 and 17.

Reply:
I can change it so that it replaces the first header.